components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java

```diff
@@ -600,14 +602,62 @@ public static String getNewOrganizationVersion() {
      */
     public static boolean isLoginAndRegistrationConfigInheritanceEnabled(String tenantDomain) {
 
+        return isOrgVersionApplicable(tenantDomain,
+                OrganizationManagementConstants.OrganizationVersion.ORG_VERSION_V1);
+    }
+
+    /**
+     * Check whether claim and OIDC claim inheritance is enabled for the organization identified by the given tenant
+     * domain.
+     *
+     * Claim and OIDC claim inheritance is considered to be enabled if the organization version is
+     * equal to or greater than v1.0.0
+     *
+     * @param tenantDomain Tenant domain of the organization.
+     * @return true if the version is v1.0.0 or above, false otherwise.
+     */
+    public static boolean isClaimAndOIDCScopeInheritanceEnabled(String tenantDomain) {
+
+        return isOrgVersionApplicable(tenantDomain,
+                OrganizationManagementConstants.OrganizationVersion.ORG_VERSION_V1);
+    }
+
+    /**
+     * Checks whether the current organization version is greater than or equal to the minimum
+     *
+     * @param tenantDomain             Tenant domain of the organization.
+     * @param minimumApplicableVersion The minimum applicable version.
+     * @return true if the org version is greater than or equal to the minimum applicable version.
+     */
+    private static boolean isOrgVersionApplicable(String tenantDomain, String minimumApplicableVersion) {
+
         try {
             String orgVersion = organizationManager.getOrganization(
                     organizationManager.resolveOrganizationId(tenantDomain), false, false).getVersion();
-            if (StringUtils.isBlank(orgVersion)) {
+            if (StringUtils.isBlank(orgVersion) || StringUtils.isBlank(minimumApplicableVersion)) {
                 return false;
             }
-
-            return orgVersion.compareTo(OrganizationManagementConstants.OrganizationVersion.ORG_VERSION_V1) >= 0;
+            // Semantic version comparison
+            String[] orgVersionComponents = orgVersion.replace(ORG_VERSION_PREFIX, StringUtils.EMPTY)
+                    .split(ORG_VERSION_DELIMITER_REGEX);
+            String[] minVersionComponents = minimumApplicableVersion.replace(ORG_VERSION_PREFIX, StringUtils.EMPTY)
+                    .split(ORG_VERSION_DELIMITER_REGEX);
+            int length = Math.max(orgVersionComponents.length, minVersionComponents.length);
+            for (int i = 0; i < length; i++) {
+                int orgVersionComponent = (i < orgVersionComponents.length) ?
+                        Integer.parseInt(orgVersionComponents[i]) : 0;
```

Comment:
Hope we don't have to handle `NumberFormatException` runtime exception here since the orgVersionComponents[i] is always a valid integer